iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/CertificateBasedAuthenticationCheck.java

```diff
@@ -74,12 +80,22 @@ private static void checkRegistriesTokens(ContextualResource resource) {
     }
   }
 
+  private static void checkLinkedServices(ContextualResource resource) {
+    ContextualProperty type = resource.property("type");
+    ContextualProperty authenticationType = resource.object("typeProperties").property("authenticationType");
+
+    if (type.is(isValue(SENSITIVE_LINKED_SERVICES_TYPE::contains))
+      && authenticationType.is(isValue(str -> !"ClientCertificate".equals(str)))) {
+      authenticationType.report(WRONG_AUTHENTICATION_METHOD_MESSAGE, type.toSecondary("Service type."));
```

Comment:
I'm not sure if the secondary location messages also contain a `.` at the end.

Reply:
I checked other secondaries, indeed there is no final `.`, it will remove it!